crates/biome_module_graph/src/resolver_cache.rs

```diff
@@ -92,11 +92,15 @@ impl<'a> ResolverCache<'a> {
     }
 
     fn path_kind(&self, path: &Utf8Path) -> Option<PathKind> {
```

Comment:
This method prevented the graph to be updated during the traversal as `added_paths` contains only the current scanning path.